### PR TITLE
Add pre-commit config

### PR DESCRIPTION
**Pull request by philippjfr**
As title says adds a pre-commit configuration file adding some basic linting rules and flake8 to pre-commit rather than running it in the linter test suite. Local setup is easy:

- Install pre-commit with `pip install pre-commit` or `conda install -c conda-forge pre-commit`
- Now run `pre-commit install` in the panel repo root directory
- Now the hooks will run before every commit

On CI the pre-commit hooks run before all the test jobs are started avoiding unnecessarily starting a bunch of jobs.

### CODE REVIEW

**Comment by maximlt** [comment]
Excellent!

**Comment by jbednar** [comment]
Can you also summarize the source code changes? What I can see on spot checks is removing trailing whitespace and adding newlines at the end of files. Anything else?

**Comment by philippjfr** [comment]
That is it as far as I'm aware. pre-commit auto fixes the failures and that's the only ones that it reported.

File: panel/models/reactive_html.py [hunk 1]
```diff
@@ -106,6 +106,7 @@ def handle_data(self, data):
         values_loop = re.findall(values_iter_re, data)
         items_loop = re.findall(items_iter_re, data)
         nloops = len(list_loop) + len(values_loop) + len(items_loop)
+        print(list_loop)
```
Comment (Hoxbro) [inline panel/models/reactive_html.py:109]: Loose print statement. Wondering if pre-commit could/should catch a print statement like this?

Comment (maximlt) [reply panel/models/reactive_html.py:109]: I've seen a pre-commit plugin that does that. There are a few other print statements in the code base, so first we'd have to check if they can be replaced by logging calls, and replace them if so.

Comment (philippjfr) [reply panel/models/reactive_html.py:109]: Thanks @Hoxbro. Can also mark them explicitly to be ignored.

**Pull request merged**